optaplanner-core/src/main/java/org/optaplanner/core/impl/score/stream/ConstraintSessionFactory.java

```diff
@@ -28,6 +28,6 @@
      * @param workingSolution if null, uniform synthetic constraint weights will be applied
      * @return never null
      */
-    ConstraintSession<Solution_, Score_> buildSession(boolean constraintMatchEnabled, Solution_ workingSolution);
+    Object buildSession(boolean constraintMatchEnabled, Solution_ workingSolution);
```

Comment:
`KieSession`/`BavetConstraintSession` have nothing in common.
Handling of this happens in `*ConstraintStreamScoreDirectorFactory` where it becomes fully type-safe.